### PR TITLE
Fix doc/module2rst.py to use glob and remove --nowarn from travis-sphinx

### LINKED ISSUE #1154: index is missing in doc
see "Indices and tables" https://espnet.github.io/espnet/

### PR DESCRIPTION
**Pull request by ShigekiKarita**
Fix #1154 

I stop using pkgutil because it does not import dir without `__init__.py`
So I will use glob instead because it literally retrieves any files under the dir.

### CODE REVIEW

**Comment by ShigekiKarita** [comment]
This is screenshot. Now modules are back.
image

**Comment by ShigekiKarita** [comment]
I found that now sphinx warning is really reduced. https://travis-ci.org/espnet/espnet/jobs/578663756#L6003-L6008

**Comment by ShigekiKarita** [comment]
I removed `--nowarn` from travis-sphinx!  Now CI report docstring warning as error.

**Comment by ShigekiKarita** [comment]
Hmm, I found `doc/module2rst.py` ignored nested files before. So warning was too short.

Now I revert the `--nowarn` and there are several critical ones like `asr_mix.py` import error. I gave up.

**Comment by kan-bayashi** [comment]
Could you upload html somewhere ?

**Comment by ShigekiKarita** [comment]
http://karita.xyz/espnet-fix-doc/

**Comment by kan-bayashi** [comment]
`__init__` is not appeared in document.
Could you try following option?
https://stackoverflow.com/questions/36270358/sphinx-autodoc-skip-member-handler-cant-show-init-when-using-napoleon

**Comment by ShigekiKarita** [comment]
It is not sphinx problem but pkgutil.

**Comment by ShigekiKarita** [comment]
see also https://stackoverflow.com/questions/41203765/init-py-required-for-pkgutil-walk-packages-in-python3

File: doc/module2rst.py [hunk 1]
```diff
@@ -60,9 +43,14 @@ def gen_rst(module, f):
...
 
-    cs = children(module, recursive=True)
-    for c in cs:
-        cname = c.__name__
+    for cpath in glob(module_path + "/**/*.py", recursive=True):
+        if not os.path.exists(cpath):
+            continue
+        if "__pycache__" in cpath:
+            continue
+        if "__init__.py" in cpath:
```
Comment (ShigekiKarita) [inline doc/module2rst.py]: @kan-bayashi I misunderstood your question. I explicitly removed the `__init__.py`. I will revert that.

**Comment by ShigekiKarita** [comment]
Do you need espnet.__init__.py document?

**Comment by ShigekiKarita** [comment]
I think `__init__.py` is only one. And it is empty
```
$ find espnet -name "__init__.py"                                
espnet/__init__.py
$ cat espnet/__init__.py
```

**Comment by kan-bayashi** [comment]
Sorry, I mean `__init__` in the class.

**Comment by ShigekiKarita** [comment]
OK

**Comment by kan-bayashi** [comment]
e.g. e2e_tts_tacotron2.__init__()

**Comment by ShigekiKarita** [comment]
fixed http://karita.xyz/espnet-fix-doc/_gen/espnet.nets.html#espnet.nets.pytorch_backend.e2e_tts_tacotron2.Tacotron2

**Pull request merged**